exporter/fileexporter/README.md

```diff
@@ -21,36 +20,98 @@ This intended for primarily for debugging Collector without setting up backends.
-    - max_days: [no default (unlimited)]: the maximum number of days to retain telemetry files based on the timestamp encoded in their filename.
-    - max_backups: [default: 100]: the maximum number of old telemetry files to retain.
-    - localtime : [default: false (use UTC)] whether or not the timestamps in backup files is formatted according to the host's local time.
+
+  - max_megabytes:  [default: 100]: the maximum size in megabytes of the telemetry file before it is rotated.
+  - max_days: [no default (unlimited)]: the maximum number of days to retain telemetry files based on the timestamp encoded in their filename.
+  - max_backups: [default: 100]: the maximum number of old telemetry files to retain.
+  - localtime : [default: false (use UTC)] whether or not the timestamps in backup files is formatted according to the host's local time.
+
+- `format`[default: json] :define the data format of encoded telemetry data. The setting can be overridden with `proto`.
+
+
 
 ## File Rotation
+
 Telemetry is first written to a file that exactly matches the `path` setting. When the file size exceeds `max_megabytes` or age exceeds `max_days`, the file will be rotated.
 
 When a file is rotated, **it is renamed by putting the current time in a timestamp**
 in the name immediately before the file's extension (or the end of the filename if there's no extension).
 **A new telemetry file will be created at the original `path`.**
 
 For example, if your `path` is `data.json` and rotation is triggered, this file will be renamed to `data-2022-09-14T05-02-14.173.json`, and a new telemetry file created with `data.json`
+
+
+
+##  File Format
+
+Telemetry data is encoded according to the `format` setting and then written to the file.
+
+When `format` is json,  telemetry data is written to file in JSON format. A line in the file is a JSON encoded data.
+
+When `format`  is proto,  encoded telemetry data can't be written row by row like in JSON format. It's hard to keep track of where one message ends and the next begins. The easiest way to solve this problem is to write the size of each message before you write the message itself.
+
+```go
+  data := make([]byte, 4, 4+len(buf))
+	binary.BigEndian.PutUint32(data, uint32(len(buf)))
+	data = append(data, buf...)
+	if err := binary.Write(e.file, binary.BigEndian, data); err != nil {
+		return err
+	}
+```
+
+When we need read the messages back in, we read the size, then read the bytes into a separate buffer, then parse from that buffer.
+
+```go
+func readMessageFromStream(br *bufio.Reader) ([]byte,  error) {
+	var length int32
+	// read length
+	err := binary.Read(br, binary.BigEndian, &length)
+	if err != nil {
+		if errors.Is(err, io.EOF) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	buf := make([]byte, length)
+	err = binary.Read(br, binary.BigEndian, &buf)
+	if err == nil {
+		return buf, nil
+	}
+	if errors.Is(err, io.EOF) {
+		return nil, nil
+	}
+	return nil, err
+}
+```
```

Comment:
This is potentially very helpful, but I think the explanation of the format is sufficient for now. We may eventually want to implement this in a complementary receiver as well.

Reply:
ok, I will remove it